File: app/src/main/java/com/vikingsen/cheesedemo/model/database/cheese/CheeseManager.java
/*
 * CheeseManager.java
 *
 * Generated on: 04/01/2017 09:45:31
 *
 */


package com.vikingsen.cheesedemo.model.database.cheese;

import com.vikingsen.cheesedemo.model.database.DatabaseManager;

import org.dbtools.android.domain.date.DBToolsThreeTenFormatter;
import org.dbtools.query.sql.SQLQueryBuilder;
import org.threeten.bp.LocalDateTime;

import java.util.List;

import rx.Observable;


@javax.inject.Singleton
public class CheeseManager extends CheeseBaseManager {

    private static final LocalDateTime DAWN_OF_TIME = LocalDateTime.of(1970, 1, 1, 0, 0);

    @javax.inject.Inject
    public CheeseManager(DatabaseManager databaseManager) {
        super(databaseManager);
    }

    public Cheese findByCheeseId(long cheeseId) {
        return findBySelection(CheeseConst.C_ID + "=?", SQLQueryBuilder.toSelectionArgs(cheeseId), null);
    }

    public LocalDateTime findOldestCacheDate() {
        LocalDateTime cached = DBToolsThreeTenFormatter.dbStringToLocalDateTime(
                findValueBySelection(String.class, CheeseConst.C_CACHED, null, null, CheeseConst.C_CACHED + " DESC", null)
        );
        if (cached == null) {
            return DAWN_OF_TIME;
        }
        return cached;
    }

    public LocalDateTime findCacheDate(long id) {
        LocalDateTime cached = DBToolsThreeTenFormatter.dbStringToLocalDateTime(
                findValueBySelection(String.class, CheeseConst.C_CACHED, CheeseConst.C_ID + "=?", SQLQueryBuilder.toSelectionArgs(id), null)
        );
        if (cached == null) {
            return DAWN_OF_TIME;
        }
        return cached;
    }

    public Observable<List<Cheese>> findAllCheesesRx() {
        return findAllOrderByRx(CheeseConst.C_NAME);
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/model/database/cheese/CheeseManager.kt
/*
 * CheeseManager.kt
 *
 * Generated on: 04/14/2017 02:45:41
 *
 */



package com.vikingsen.cheesedemo.model.database.cheese

import com.vikingsen.cheesedemo.model.database.DatabaseManager
import io.reactivex.Maybe
import io.reactivex.Single
import org.dbtools.android.domain.date.DBToolsThreeTenFormatter
import org.dbtools.query.sql.SQLQueryBuilder
import org.threeten.bp.LocalDateTime
import javax.inject.Inject


@javax.inject.Singleton
class CheeseManager
@Inject constructor(databaseManager: DatabaseManager) : CheeseBaseManager(databaseManager) {

    fun findByCheeseIdRx(cheeseId: Long): Maybe<Cheese> {
        return findBySelectionRx(
                selection = "${CheeseConst.C_ID} = ?",
                selectionArgs = SQLQueryBuilder.toSelectionArgs(cheeseId)
        )
    }

    fun findOldestCacheDate(): LocalDateTime {
        val cached = DBToolsThreeTenFormatter.dbStringToLocalDateTime(
                findValueBySelection(
                        valueType = String::class.java,
                        column = CheeseConst.C_CACHED,
                        defaultValue = null,
                        orderBy = "${CheeseConst.C_CACHED} DESC"
                )
        ) ?: return DAWN_OF_TIME
        return cached
    }

    fun findCacheDate(id: Long): LocalDateTime {
        val cached = DBToolsThreeTenFormatter.dbStringToLocalDateTime(
                findValueBySelection(
                        valueType = String::class.java,
                        column = CheeseConst.C_CACHED,
                        defaultValue = null,
                        selection = "${CheeseConst.C_ID} = ?",
                        selectionArgs = SQLQueryBuilder.toSelectionArgs(id)
                )
        ) ?: return DAWN_OF_TIME
        return cached
    }

    fun findAllCheesesRx(): Single<List<Cheese>> {
        // GOTCHA - MATCH ORDERING FROM SERVER
        return findAllRx(orderBy = CheeseConst.C_NAME)
    }

    companion object {
        private val DAWN_OF_TIME = LocalDateTime.of(1970, 1, 1, 0, 0)
    }
}